### ISSUE TITLE
Clean up code following the removal of _oldtls

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7051]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Alex reported|
|-|-|
|Trac ID|trac#7051|
|Type|enhancement|
|Created|2014-03-24 23:14:30Z|
|Branch|https://github.com/twisted/twisted/tree/no-getattr-please-7051|

There's some stuff left over which is now just cruft.

Attachments:

* [t7051.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/386/3869cc25a6afad5b51c669bcf0bfafdaeb6ca951/6635ad73886bdc8d511b37ea9469ff3d3e016271.diff) (2341 bytes) - added by Alex on 2014-03-24 23:15:28Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__7051 7051
type__enhancement enhancement
reporter__Alex Alex
priority__normal normal
milestone__ 
branch__branches_no_getattr_please_7051 branches/no-getattr-please-7051
branch_author__glyph glyph
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1395702870000000 1395702870000000
changetime__1395965198000000 1395965198000000
version__None None
owner__radix radix

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph commented|
|-|-|

(In [41991]) Branching to no-getattr-please-7051.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph set owner to Alex|
|-|-|

Thanks Alex,

  1. Needs a NEWS file (although probably just an empty .misc).
  2. More importantly though, the new instance variable needs maintainer documentation in the form of an `@ivar`.
  3. The buildbots are taking their sweet time completing the build, and when I ran it locally the tests hung.  This change doesn't seem to be as innocuous as it appears.  Did you run the tests locally?

**Comment by twisted-trac** [comment]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|Alex commented|
|-|-|

Fixes for Glyph's review items: http://bpaste.net/show/193273/ (sent to glyph out of band which is why it's on a pastebin).

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph commented|
|-|-|

It occurs to me we ought to be using the `socket.SHUT_*` etc. constants in this code.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph commented|
|-|-|

Looks basically OK so far but given that there are IOCP changes here I am waiting for the full buildbot quorum.

**Comment by twisted-trac** [comment]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|Alex commented|
|-|-|

Tests seem to be passing.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_8"></a>|@radix set owner to @radix|
|-|-|

Looks good, thanks. +1

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_9"></a>|@radix set status to `closed`|
|-|-|

(In [42000]) Merge no-getattr-please-7051

Author: alex
Reviewer: radix
Fixes: [#7051](https://github.com/twisted/twisted/issues/7051)

Clean up unnecessary code which would dynamically determine the
method to shut down a socket.
